addons/BattleStations/battlestations.lua

```diff
@@ -0,0 +1,598 @@
+    local packet = _packets.parse('incoming', data)
+    return {
+        day = packet['Day Music'],
+        night = packet['Night Music'],
+        solo = packet['Solo Combat Music'],
+        party = packet['Party Combat Music']
+    }
+end
+
+function getConditionalSongTranslation(song)
+    local zone_bgm_table = getZoneBGMTable()
+    if song == music.songs.others.zone then    
+        if timeIsDaytime() then
+            song = zone_bgm_table.day
+        else 
+            song = zone_bgm_table.night
+        end
+       
+        if playerInReive() then
+            song = music.songs.seekers_of_adoulin.breaking_ground
+        end
+          
+    elseif song == music.songs.others.normal then
+        if playerInParty() then
+            song = zone_bgm_table.party
+        else 
+            song = zone_bgm_table.solo
+        end
+    end
+    return song 
+end
+
+function getPlayerBuffs()
+    return T(windower.ffxi.get_player().buffs)
+end
+
+function timeIsDaytime()
+    local current_time = tonumber(getCurrentTime())
+    return current_time >= 6*60 and current_time <= 18*60
+end
+
+function playerIsFighting()
+    return windower.ffxi.get_player().status == player.statuses.fighting
+end
+
+function playerInParty() 
+    return windower.ffxi.get_party().alliance_count > 1
+end
+
+function playerInReive()
+    return getPlayerBuffs():contains(player.buffs.reiveMark)
+end
+
+function frequencyValid(frequency)
+    return stations.frequencies[tostring(frequency)] ~= nil
+end
+
+function radioValid(radio)
+    return stations.receivers[radio] ~= nil or radio == '*'
+end
+
+function categoryValid(category) 
+    return stations.categories[category] ~= nil
+end
+
+function listTypeValid(list_type) 
+    return help.lists[list_type] ~= nil
+end
+
+function handleInjectionNeeds() 
+	if needs_inject then
+		injectBattleMusic()
+		needs_inject = false;
+    end
+end
+
+windower.register_event('load', function () 
+    injectBattleMusic()
+end)
+
+windower.register_event('unload', function() 
+    local music_type = music.types.battle_solo
+    local zone_bgm_table = getZoneBGMTable()
+    local song = zone_bgm_table.solo
+    
+    if playerInParty() then
+       music_type = music.types.battle_party
+       song = zone_bgm_table.solo
+    end
+                   
+    _packets.inject(_packets.new('incoming', packets.inbound.music_change.id, {
+        ['BGM Type'] = music_type,
+        ['Song ID'] = song,
+    }))
+end)
+
+windower.register_event('action', function(act)
+    if act.actor_id == windower.ffxi.get_player().id then
+        if act.category == 4 and act.recast == 225 and act.targets[1].actions[1].animation == 939 then
+            if not playerInParty() then
+                functions.loop(injectBattleMusic, 1, 5)          
+            end    
+        end
+    end
+end)
+
+windower.register_event('outgoing chunk', function(id, data)
+	if id == packets.outbound.action.id then
```

Comment:
Another tab... it may make sense to just do a search and replace, looks like it happens occasionally.

Reply:
done